src/coord/src/sink_connector.rs

```diff
@@ -38,6 +43,143 @@ pub async fn build(
     }
 }
 
+/// Polls a message from a Kafka Source
+fn get_next_message(
+    consumer: &mut BaseConsumer,
+    timeout: Duration,
+) -> Result<Option<Vec<u8>>, anyhow::Error> {
+    if let Some(result) = consumer.poll(timeout) {
+        match result {
+            Ok(message) => match message.payload() {
+                Some(p) => Ok(Some(p.to_vec())),
+                None => {
+                    bail!("unexpected null payload")
```

Comment:
Is this also fine with compacted topics? (haven't read the rest of the code yet so maybe a dumb question)

Reply:
edit: after reading further this is for the consistency topic and so it shouldn't be compacted.